### PR TITLE
[WIP] CLC driver

### PR DESCRIPTION
**Pull request by ack**
### Status
- work in progress
- looking for initial feedback
### Checklist (tick everything that applies)
- [ ] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [ ] Documentation
- [ ] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [ ] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by Kami** [comment]
Great, thanks!

Please keep us posted on the progress and let us know when it's ready to be progress.

So far it looks like a good start.

**Comment by Kami** [comment]
In the mean time, since this is a bigger change, please make sure to file an ICLA - http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes

File: libcloud/common/clc.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        Adds vendor headers:
+        - authorization header when token present
+        - api client
+        """
+        if self._token:
+            headers['Authorization'] = 'Bearer %s' % (self._token)
+        headers['Content-Type'] = 'application/json'
+        headers['Api-Client'] = self._user_agent()
+        return headers
+
```
Comment (tonybaloney) [inline libcloud/common/clc.py:109]: remove extra line break

File: libcloud/compute/drivers/clc.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        path = '/v2/servers/%s/%s' % (self.alias, id)
+        resp = self.connection.request(path)
+        return self._to_node(resp.object)
+
+    def list_locations(self):
+        path = '/v2/datacenters/%s' % (self.alias)
+        resp = self.connection.request(path)
+        return map(self._to_location, resp.object)
+
+    def get_location(self, dc, groups=False):
```
Comment (tonybaloney) [inline libcloud/compute/drivers/clc.py:82]: please keep argument names inline with the base class or prepend with `ex_`

File: libcloud/compute/drivers/clc.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        dc = self._get_dc(dc)
+        return self._to_location(dc)
+
+    def list_images(self, dc):
+        # images == templates
+        obj = self.ex_deployment_capabilities(dc)
+        return map(self._to_image, obj['templates'])
+
+    def list_nodes(self, gid):
```
Comment (tonybaloney) [inline libcloud/compute/drivers/clc.py:91]: methods need docstrings

File: libcloud/compute/drivers/clc.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        for l in dc['links']:
+            if l['rel'] == 'group':
+                groups.append(dict(id=l['id'], name=l['name']))
+        return groups
+
+    def ex_deployment_capabilities(self, dc):
+        path = '/v2/datacenters/%s/%s/deploymentCapabilities' % \
+               (self.alias, dc)
+        resp = self.connection.request(path)
+        return resp.object
```
Comment (tonybaloney) [inline libcloud/compute/drivers/clc.py:117]: if you can create a more concrete model this would be advisable

**Comment by tonybaloney** [comment]
apart from some style comments, the driver definitely needs a test suite as well as a supporting collection of fixtures from actual CLC API endpoints. Then it needs some documentation in the `docs/` folder, there are plenty of examples from other drivers.

**Comment by Kami** [comment]
@ack How is this looking?

It would be great if we can get it finished and merged...

**Pull request closed without merging**